### PR TITLE
Adding support for Service Hooks generator

### PR DESCRIPTION
**Pull request by hkamel**
Thank you for your pull request!
By completing this pull request, you agree to the [Contributing License Agreement](https://github.com/ALM-Rangers/Yeoman-extension-code-generator/blob/master/.github/CLA.md).

Fixes # .

Changes proposed in this pull request:  
-  Adding support for Service Hooks generator

@ALM-Rangers/Yeoman-extension-code-generator

### CODE REVIEW

File: generators/app/index.js [hunk 1]
```diff
@@ -55,6 +55,10 @@ function input() {
             {
                name: 'Widget dashboard',
                value: 'widget'
+            },
+            {
+                name: 'Service hook',
```
Comment (willsmythe) [inline generators/app/index.js]: I would call this a "Service hooks consumer" so it's clear what is being contributed.

File: README.md [hunk 2]
```diff
@@ -12,6 +12,7 @@ The extension type can be :
 - New hub extension
 - Custom build or release task
 - Widget dashboard
+- Service Hooks Consumer
```
Comment (willsmythe) [inline README.md]: "Service hooks consumer" (to match the case of others)

File: generators/app/utility.js [hunk 3]
```diff
@@ -107,6 +107,28 @@ function setJsonTaskScript(scriptMode, taskname) {
       return jsonproperty;
 }
 
+function setEventsText(eventsList) {
+      
+            var jsonproperty = {
```
Comment (willsmythe) [inline generators/app/utility.js]: Formatting problems. Maybe run code format to fix up.

File: generators/servicehook/index.js [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+}
+
+function input() {
+
+      // Collect any missing data from the user.
+      // This gives me access to the generator in the
+      // when callbacks of prompt
+      let cmdLnInput = this;
+
+      return this.prompt([
```
Comment (willsmythe) [inline generators/servicehook/index.js:24]: FUTURE: should pull this common stuff (that's common across all generators) into a single class so you don't have to repeat the same input definitions over and over (and it's easier to change in one place later).

File: generators/servicehook/index.js [hunk 5]
```diff
@@ -0,0 +1,258 @@
...
+                  message: 'Please enter a description of your extension:',
+                  default: 'My extension makes hubs great again',
+                  when: answers => {
+                        return cmdLnInput.extDescription === undefined;
+                  }
+            }, {
+                  type: 'input',
+                  name: 'publisherId',
+                  store: true,
+                  message: 'Please enter the name of your marketplace publisher ID:',
```
Comment (willsmythe) [inline generators/servicehook/index.js]: uppercase "M" on Marketplace .. might even spell it out "Visual Studio Marketplace"

File: generators/servicehook/index.js [hunk 6]
```diff
@@ -0,0 +1,258 @@
...
+                  validate: util.validateExtensionName,
+                  default: 'My Extension',
+                  when: answers => {
+                        return cmdLnInput.extName === undefined;
+                  }
+            }, {
+                  type: 'input',
+                  name: 'extId',
+                  store: true,
+                  message: 'Please enter the ID of your extension:',
```
Comment (willsmythe) [inline generators/servicehook/index.js]: Nit (should have caught this before) ... " .. enter an ID for your .. "

File: generators/servicehook/templates/ExtId.csproj [hunk 7]
```diff
@@ -0,0 +1,94 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (willsmythe) [inline generators/servicehook/templates/ExtId.csproj:1]: It'll be nice to switch at some point to the new 2017 .csproj format. Soooo much nicer.

File: README.md (hunk 2 above)
Comment (hkamel) [reply README.md]: done

**Review by hkamel — COMMENTED** [review]
Thanks @willsmythe I have accommodated most of the comments and updated the PR

File: generators/servicehook/index.js (hunk 4 above)
Comment (wpschaub) [reply generators/servicehook/index.js:24]: @mikaelkrief, can we put this recommendation on our backlog?

File: generators/servicehook/templates/ExtId.csproj (hunk 7 above)
Comment (wpschaub) [reply generators/servicehook/templates/ExtId.csproj:1]: @mikaelkrief, can we put this recommendation on our backlog?

File: generators/servicehook/index.js [hunk 8]
```diff
@@ -55,7 +55,7 @@ function input() {
                   type: 'input',
                   name: 'publisherId',
                   store: true,
-                  message: 'Please enter the name of your marketplace publisher ID:',
+                  message: 'Please enter the name of your Marketplace publisher ID:',
```
Comment (wpschaub) [inline generators/servicehook/index.js]: @mikaelkrief, can we put this change on our backlog to apply to other tasks for consistency, i.e. "Marketplace publisher"?

File: generators/servicehook/index.js [hunk 9]
```diff
@@ -36,7 +36,7 @@ function input() {
                   type: 'input',
                   name: 'extId',
                   store: true,
-                  message: 'Please enter the ID of your extension:',
+                  message: 'Please enter an ID of your extension:',
```
Comment (wpschaub) [inline generators/servicehook/index.js]: @mikaelkrief, can we put this change on our backlog to apply to other tasks for consistency, i.e. "an ID"?

File: generators/servicehook/index.js (hunk 4 above)
Comment (mikaelkrief) [reply generators/servicehook/index.js:24]: yes,
Agree.

File: generators/servicehook/templates/ExtId.csproj (hunk 7 above)
Comment (mikaelkrief) [reply generators/servicehook/templates/ExtId.csproj:1]: yes

**Review by mikaelkrief — APPROVED** [review]

**Review by wpschaub — APPROVED** [review]

**Pull request merged**